### PR TITLE
Improve the performance of the linkcheck list

### PR DESCRIPTION
**Pull request by david-venhoff**
### Short description

This pr reduces the loading time when opening the link list view. More performance optimizations are possible, but would required bigger changes.
This pr also fixes a bug that the list view basically did not load at all when the django debug toolbar was activated.

### Proposed changes

- Filter urls in sql instead of in python
- Don't use the prefetch optimization, which does not appear to be worth it

### Side effects

- Due to not using the prefetch optimization the total number of database queries for the link list has increased 

### Resolved issues

Fixes: #2539 (Maybe? I could never reproduce the issue where loading takes 20 seconds. Even before this pr I 'only' had to wait for 10 seconds at most)

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

File: integreat_cms/cms/utils/linkcheck_utils.py [hunk 1]
```diff
@@ -55,27 +52,27 @@ def get_urls(
...
     # Filter out ignored URL types
-    urls = [url for url in urls if url.type not in settings.LINKCHECK_IGNORED_URL_TYPES]
-    if region_slug:
-        # If the region slug is given, only return urls that occur at least once in the requested region
-        urls = [url for url in urls if url.region_links]
+    if settings.LINKCHECK_IGNORED_URL_TYPES:
+        urls = [
+            url for url in urls if url.type not in settings.LINKCHECK_IGNORED_URL_TYPES
+        ]
     return urls
```
Comment (deen13) [inline integreat_cms/cms/utils/linkcheck_utils.py:72]: Returning either the filtered array or the urls variable reduces state mutation and might make the code easier to understand, imo. :)

```suggestion
    if settings.LINKCHECK_IGNORED_URL_TYPES:
        return [url for url in urls if url.type not in settings.LINKCHECK_IGNORED_URL_TYPES]
    else:
        return urls
```

**Review by deen13 — COMMENTED** [review]
Thanks for the performance improvement, @david-venhoff. It looks good to me code-wise! I'm not entirely sure how to test the performance optimization, though. Do we have a large dataset that we can use for a test case?

**Comment by david-venhoff** [comment]
I think the currently recommended way to test the performance is to load a database dump into your dev-setup.
If you don't have permissions to accesses the testserver db, I can send you the dump I've been using for writing this pr :)

**Comment by deen13** [comment]
> I think the currently recommended way to test the performance is to load a database dump into your dev-setup.
> If you don't have permissions to accesses the testserver db, I can send you the dump I've been using for writing this pr :)

That'd be great! @david-venhoff

**Comment by deen13** [comment]
### Testing with Database Dump (Docker Version)

Follow these steps to set up your environment for testing the pull request using a production database dump. 

#### Step 1: Obtain the Database Dump
1. Request access from @david-venhoff.
2. Download the dump from https://nextcloud.tuerantuer.org/index.php/f/6459275
3. Unzip the dump (`gzip -d integreat-2024-05-25.sql.gz`)

#### Preparing application
- Ensure your container volume is empty (`docker volume rm <volume_id>`)
- Start the Docker container without migration (`docker run -d --name integreat_django_postgres <other_options>`)
- Import the database dump (`cat integreat-2024-05-25.sql | docker exec -i integreat_django_postgres psql -U integreat`)
- Remove everything from test_data.json except the admin user
- Load the root user to re-grant access (`./tools/loadtestdata.sh`)

You can now log in with the default credentials and have the test data in place. Happy testing! 🤗

**Review by deen13 — APPROVED** [review]
The changes improved the loading time in the regions krefeld and bocholt from not loading at all to 1.8 seconds on my machine. 🙈 😄

**Comment by david-venhoff** [comment]
>     * Remove everything from test_data.json except the admin user
> 
>     * Load the root user to re-grant access (`./tools/loadtestdata.sh`)

Alternatively, it is possible to instead call `./tools/create_superuser.sh` :)

**Review by MizukiTemma — APPROVED** [review]
Wow 😻 It's now way faster 😻😻 So cool 😻😻😻

You can probably add a release note. This doesn't change visible functionalities but brings an awesome improvement :)

**Comment by MizukiTemma** [comment]
> ### Testing with Database Dump (Docker Version)
> Follow these steps to set up your environment for testing the pull request using a production database dump.
> 
> #### Step 1: Obtain the Database Dump
> * Request access from @david-venhoff.
> * Download the dump from https://nextcloud.tuerantuer.org/index.php/f/6459275
> * Unzip the dump (`gzip -d integreat-2024-05-25.sql.gz`)
> 
> #### Step 2: Prepare the Application
> * Ensure your container volume is empty (`docker volume rm <volume_id>`)
> * Start the Docker container without migration (`docker run -d --name integreat_django_postgres <other_options>`)
> * Import the database dump (`cat integreat-2024-05-25.sql | docker exec -i integreat_django_postgres psql -U integreat`)
> * Remove everything from test_data.json except the admin user
> * Load the root user to re-grant access (`./tools/loadtestdata.sh`)
> 
> You can now log in with the default credentials and have the test data in place. Happy testing! 🤗

Thank you for the how-to 😉

**Pull request merged**